source/common/buffer/watermark_buffer.h

```diff
@@ -35,25 +37,29 @@ class WatermarkBuffer : public OwnedImpl {
   Api::IoCallUint64Result write(Network::IoHandle& io_handle) override;
   void postProcess() override { checkLowWatermark(); }
 
-  void setWatermarks(uint32_t watermark) { setWatermarks(watermark / 2, watermark); }
-  void setWatermarks(uint32_t low_watermark, uint32_t high_watermark);
+  void setWatermarks(uint32_t watermark) { setWatermarks(watermark / 2, watermark, watermark * 2); }
```

Comment:
I'm not certain about this... I need to do some digging to better understand how these buffers are created in different contexts. My understanding is that `void setWatermarks(uint32_t watermark)` provides a convenient default behavior, and that `void setWatermarks(uint32_t low_watermark, uint32_t high_watermark, uint32_t overflow_watermark)` gives you full control, but I haven't yet looked into which variant is called where.

Reply:
the network connection calls setWatermarks() under setBufferLimits() so it's going to end up with 2x.  I think it might be safest for us to apply that to the network::connection in H2 codec creation, so any codec path be it upstream, downstream, health check etc, gets the right limits.

I'd also lean towards runtime featrure guarding this even if you have it on by default, so folks can easily turn it off if it causes problems.  The simplest way to guard would probably just only calling the new overflow callback if the feature is on, because that way it disables all the various close / reset points.

I don't think you can do that until #7695 lands, and it's worth getting Matt's call if I'm being overly paranoid too :-)

Reply:
+1 for runtime control. One option would be to actually have the multiplier pulled from runtime, and have 0 turn it off, but having a boolean and cancelling the callbacks is also fine.

Reply:
Just getting back to this. I like @mattklein123's idea of a multiplier param that defaults to 0. So of course the hardest and most important question now is, what should the runtime key be called?

Reply:
My strawman: `buffer.overflow.high_watermark_multiplier`.

Reply:
That SGTM, but I would probably default to on at 2 (or whatever we decide) and then allow people to turn it off with 0.